### ISSUE TITLE
Move to Maven/Tycho for projects and build infrastructure

### ISSUE THREAD

**Issue by LorenzoBettini** [eclipse/xtext-umbrella#350]
See also https://github.com/eclipse/xtext-lib/issues/499 and https://github.com/eclipse/xtext-core/issues/2052 https://github.com/eclipse/xtext-extras/issues/858 https://github.com/eclipse/xtext-xtend/issues/1445 https://github.com/eclipse/xtext-eclipse/issues/1992

**Comment by LorenzoBettini** [comment]
@cdietrich would it be OK if I move all the projects of `releng` as top-level projects? Or do you think that before repo merging it's better to leave them where they are?

**Comment by cdietrich** [comment]
i dont care for now.

**Comment by LorenzoBettini** [comment]
@cdietrich for the moment I won't touch them.

@merks , as usual, could you please check that this p2 repository is ok?

https://ci.eclipse.org/xtext/job/xtext-umbrella/job/lb_2052_maven_tycho/lastSuccessfulBuild/artifact/build/p2-repository/

This should be the aggregated Xtext repository with signed contents (the signed bundles are taken from the previously built p2 repositories in Jenkins jobs of the single Git repositories). However, the signing of the features takes place in the Jenkins job of this Git repository.

@cdietrich there are a few sources missing because I had changed the feature layout in some features like `.xbase.lib`; if you remember, I said that in general a feature should not contain sources of bundles. We can then later decide about that: revert the old behavior or update all features accordingly (but then also publish source features in the p2 repository)

**Comment by LorenzoBettini** [comment]
@merks you mentioned that you use a Jenkins job that we can copy in our JIRO, https://ci.eclipse.org/oomph/job/repository-analyzer-extra/

if we do that, I guess we should change the output dir of the report, or can we still generate the result in the same area where you generate your reports?

**Comment by merks** [comment]
@LorenzoBettini 

I've regenerated the report:

https://download.eclipse.org/oomph/archive/reports-extra/xtext-tycho/ci.eclipse.org/xtext/job/xtext-core/job/lb_2052_maven_tycho/lastSuccessfulBuild/artifact/build/p2-repository/index.html

Yes, if you copy the job you will have to change the where things are published.  I changed it a bit to make it easier to copy and paste it.  This section needs to change:
```
# Project-specific locations.
# Remember use to "archive" in the name of the output so that the reports are not mirrored.
#
LOGIN="genie.oomph@projects-storage.eclipse.org"
OUT_DIR="/home/data/httpd/download.eclipse.org/oomph/archive"
OUTPUT="reports-extra/${NAME}"
```

The URL and NAME parameters on the job you may wish to hard-code...

**Comment by merks** [comment]
https://download.eclipse.org/oomph/archive/reports-extra/xtext-umbrella-tycho/ci.eclipse.org/xtext/job/xtext-umbrella/job/lb_2052_maven_tycho/lastSuccessfulBuild/artifact/build/p2-repository/index.html

**Comment by LorenzoBettini** [comment]
@merks Thanks!
Just to be sure: the report says everything is OK, doesn't it? I mean from the signing point of view (which is what @cdietrich wants to be sure of).

**Comment by merks** [comment]
Yes, if something were not signed it would show up in the signing certificate sections.  Note too that the job status shows if there are serious problems:

https://ci.eclipse.org/oomph/job/repository-analyzer-extra/

There are actual test results:

https://ci.eclipse.org/oomph/job/repository-analyzer-extra/124/testReport/

**Comment by LorenzoBettini** [comment]
> Yes, if something were not signed it would show up in the signing certificate sections. Note too that the job status shows if there are serious problems:
> 
> https://ci.eclipse.org/oomph/job/repository-analyzer-extra/
> 
> There are actual test results:
> 
> https://ci.eclipse.org/oomph/job/repository-analyzer-extra/124/testReport/

Thank you so much for the great tools and support! :)

**Comment by LorenzoBettini** [comment]
> i dont care for now. but i guess merging wll be easier if same structure

OK, so I'll move all the projects at the root level

**Comment by cdietrich** [comment]
with repo merge this is done
